src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -369,6 +374,72 @@ public void canGetAllHoldings()
     assertThat(allHoldings.stream().anyMatch(filterById(thirdHoldingId)), is(true));
   }
 
+  @Test
+  public void cannotPageWithNegativeLimit() throws Exception {
+    UUID instanceId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    holdingsClient.create(new HoldingRequestBuilder()
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)).getId();
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+
+    client.get(holdingsStorageUrl("?limit=-3"), StorageTestSuite.TENANT_ID,
+      ResponseHandler.text(getCompleted));
+
+    Response response = getCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody().trim(),
+      is("'limit' parameter is incorrect. parameter value {-3} is not valid: must be greater than or equal to 0"));
+  }
+
+  @Test
+  public void cannotPageWithNegativeOffset() throws Exception {
+    UUID instanceId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    holdingsClient.create(new HoldingRequestBuilder()
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)).getId();
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+
+    client.get(holdingsStorageUrl("?offset=-3"), StorageTestSuite.TENANT_ID,
+      ResponseHandler.text(getCompleted));
+
+    Response response = getCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody().trim(),
+      is("'offset' parameter is incorrect. parameter value {-3} is not valid: must be greater than or equal to 0"));
+  }
+
+  @Test
+  public void cannotDeleteHoldingWhenLangParameterIsInvalid() throws Exception {
```

Comment:
What makes `eng` an invalid value for language?

Reply:
it have to be a two symbol i.e. `en`

Reply:
Thanks